common/build/eslint-config-fluid/src/test/enforce-no-restricted-tags-imports.test.js

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+const { RuleTester } = require("eslint");
+const noRestrictedTagsImports = require("../custom-rules/no-restricted-tags-imports");
+
+const ruleTester = new RuleTester({
+	parser: require.resolve("@typescript-eslint/parser"),
+});
+
+const validCodeWithJSDoc = `
+/**
+ * @public
+ */
+export function publicFunction() {}
+
+import { publicFunction } from "./someModule";
+`;
+
+const validCode2WithJSDoc = `
+/**
+ * @internal
+ */
+export function internalExceptionFunction() {}
+
+import { internalExceptionFunction } from "./foo";
+`;
+
+const invalidCodeWithJSDoc = `
+/**
+ * @internal
+ */
+export function internalFunction() {}
+
+import { internalFunction } from "./internalModule";
+`;
+
+ruleTester.run("no-restricted-tags-imports", noRestrictedTagsImports, {
+	// Checks cases that should pass
+	valid: [
+		{
+			code: validCodeWithJSDoc,
+			// Valid import
```

Comment:
The comment feels like it wants to describe the test, but if so I think it makes more sense to move it to above the full object inside the `valid` array (same for the ones below). And how about making it more explicit? Something like `Importing something with a non-restricted tag`.